service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java

```diff
@@ -424,4 +426,29 @@ private String stripIllegalCharacters(String str) {
         return str.replace("<", "")
             .replace(">", "");
     }
+
+    @SuppressWarnings("unchecked")
+    public Map<String, Object> fixOrderTypeTypo(String migrationId, CaseDetails caseDetails) {
+        String invalidOrderType = "EDUCATION_SUPERVISION__ORDER";
+        String validOrderType = "EDUCATION_SUPERVISION_ORDER";
+
+        Optional<Map> orders = Optional.ofNullable((Map) caseDetails.getData().get("orders"));
```

Comment:
Wonder if instead of a map might be able to use an ObjectMapper `.convertValue(field, Orders.class)`, might make all the Optionals + casting slightly more manageable! 🙂